command/monitor_test.go

```diff
@@ -0,0 +1,103 @@
+package command
+
+import (
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/vault/helper/testhelpers"
+	"github.com/mitchellh/cli"
+)
+
+func testMonitorCommand(tb testing.TB) (*cli.MockUi, *MonitorCommand) {
+	tb.Helper()
+
+	ui := cli.NewMockUi()
+	return ui, &MonitorCommand{
+		BaseCommand: &BaseCommand{
+			UI: ui,
+		},
+	}
+}
+
+func TestMonitorCommand_Run(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name string
+		args []string
+		out  string
+		code int64
+	}{
+		{
+			"valid",
+			[]string{
+				"-log-level=debug",
+			},
+			"",
+			0,
+		},
+		{
+			"too_many_args",
+			[]string{
+				"-log-level=debug",
+				"foo",
+			},
+			"Too many arguments",
+			1,
+		},
+		{
+			"unknown_log_level",
+			[]string{
+				"-log-level=haha",
+			},
+			"HAHA is an unknown log level",
+			1,
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			client, closer := testVaultServer(t)
+			defer closer()
+
+			var code int64
+			shutdownCh := make(chan struct{})
+			stopCh := make(chan struct{})
+
+			ui, cmd := testMonitorCommand(t)
+			cmd.client = client
+			cmd.ShutdownCh = shutdownCh
+
+			go testhelpers.GenerateDebugLogs(t, stopCh, client)
+			go func() {
+				atomic.StoreInt64(&code, int64(cmd.Run(tc.args)))
+			}()
+
+			// This is possibly just an artifact of how this test
```

Comment:
I'd like to work with you to eliminate this magic number, or make it nonmagic anyway.  I wonder if part of what you were seeing was because of the size 512 channel buffer?

Reply:
That's entirely possible.  Thank you!